docs/howto/system_benchmarking.md

```diff
@@ -0,0 +1,245 @@
+along with values for package and data stream-level variables. These are the available configuration options for system benchmarks.
+
+| Option | Type | Required | Description |
+|---|---|---|---|
+| package | string | | The name of the package. If omitted will pick the current package, this is to allow for future definition of benchmarks outside of the packages folders. |
+| description | string | | A description for the scenario. |
+| version | string | | The version of the package to benchmark. If omitted will pick the current version of the package. |
+| input | string | yes | Input type to test (e.g. logfile, httpjson, etc). Defaults to the input used by the first stream in the data stream manifest. |
+| vars | dictionary |  | Package level variables to set (i.e. declared in `$package_root/manifest.yml`). If not specified the defaults from the manifest are used. |
+| data_stream.name | string | yes | The data stream to benchmark. |
+| data_stream.vars | dictionary |  | Data stream level variables to set (i.e. declared in `package_root/data_stream/$data_stream/manifest.yml`). If not specified the defaults from the manifest are used. |
+| warmup_time_period | duration |  | Warmup time period. All data prior to this period will be ignored in the benchmark results. |
+| benchmark_time_period | duration |  | Amount of time the benchmark needs to run for. If set the benchmark will stop after this period even though more data is still pending to be ingested. |
+| wait_for_data_timeout | duration |  | Amount of time to wait for data to be present in Elasticsearch. Defaults to 10m. |
+| corpora.generator.size | string |  | String describing the amount of data to generate. Example: `20MiB` |
+| corpora.generator.template.raw | string |  | Raw template for the corpus generator. |
+| corpora.generator.template.path | string |  | Path to the template for the corpus generator. |
+| corpora.generator.template.type | string |  | Type of the template for the corpus generator. Default `placeholder`. |
+| corpora.generator.config.raw | dictionary |  | Raw config for the corpus generator. |
+| corpora.generator.config.path | string |  | Path to the config for the corpus generator. |
+| corpora.generator.fields.raw | dictionary |  | Raw fields for the corpus generator. |
+| corpora.generator.fields.path | string |  | Path to the fields for the corpus generator. |
+| corpora.input_service.name | string |  | Name of the input service to use (defined in the `deploy` folder). |
+| corpora.input_service.signal | string |  | Signal to send to the input service once the benchmark is ready to start. |
+
+Example:
+
+`100mb-logs-benchmark.yml`
+```yaml
+---
+description: Benchmark 100MiB of data ingested
+input: filestream
+vars: ~
+data_stream.name: test
+data_stream.vars.paths:
+  - "{{SERVICE_LOGS_DIR}}/corpus-*"
+warmup_time_period: 10s
+corpora.generator.size: 100MiB
+corpora.generator.template.path: ./100mb-logs-benchmark/template.log
```

Comment:
You mean always look for these files inside a folder named as the benchmark by convention? If that is the case I think we could since would be less error prone I guess

Reply:
++

Reply:
if we use fixed filenames we loose the possibility to have several scenarios in the same packages (for example in the case of rally benchmarks for different datasets), unless we have dynamically generated filenames

Reply:
I'm ok with this. Lets get a version in and then iterate on it.